### DISCUSSION TITLE
How do I type the items in a `FieldArray`

### DISCUSSION THREAD

**Discussion by trivedi** [react-hook-form/react-hook-form#10676] [Q&A]
I have a array that consists of object.

```js
type Student = {
  name: string,
  age: number,
}

const { control } = useForm();

const studentsArray = useFieldArray<Student>({ control, name: "students" });
```

I think I need to use `FieldArrayWithId` in the type somehow, but I can't figure it out.

If I don't type it out, tyepscript will give me an error that properties like `name` and `age` don't exist when I try to access them

**Comment by leapful** [comment]
@trivedi 

The type must be set on `useForm` hook then `useFieldArray` will infer by using `name` and `control`.

Example: https://codesandbox.io/s/calc-i231d
